### ISSUE TITLE
Pre / Post processors

### ISSUE THREAD

**Issue by Jab2870** [glacambre/firenvim#813]
- OS Version: linux x86-64
- Browser Version:  Firefox 84.0
- Browser Addon Version: 0.1.32
- Neovim Plugin Version: 0.1.32

Another feature request. I fully expect this to be out of scope for this plugin
but if you don't ask, you don't get. ;)

What are your thoughts on potentially running post / pre processors?

My example use case is that I would like to be able to write emails in markdown
in vim, but between quitting in vim and sending the contents back to the
browser, I would want to run it through something like Pandoc to convert the
markdown to html.

Similarly, to edit the email, I'd need to run a pandoc command to convert the
html back to markdown before opening it in vim.

**Comment by glacambre** [comment]
I think this could be achieved with BufEnter and BufWritePre (and maybe BufWritePost to undo changes done by BufWritePre), so I'm not sure anything is needed from Firenvim here :)

**Comment by Jab2870** [comment]
Thanks for your response. That would probably work although I think it would
depend on #812 which I raised just before this issue.

**Comment by glacambre** [comment]
I don't think this requires #812 - this is completely independent as far as I can see. Pre and post processing do not require that Firenvim does anything, everything can be done from Neovim. Or am I missing something?

**Comment by Jab2870** [comment]
Ok, the pre-processing could be done without #812 but without it, the use case
described wouldn't work because the HTML would be escaped.

I will close this issues because you are right, the processing could be done in
nvim

**Comment by jasonhildebrand** [comment]
@Jab2870 I'm interested in the same use case.  Did you get it working?  If so, could you please post the related configuration you used to do the processing?

**Comment by Jab2870** [comment]
@jasonhildebrand 

I didn't really implement this in the end using firevim. I tend to just pipe the
contents of my buffer to pandoc to convert markdown to html. Something like:

```
:%! pandoc -f markdown -t html5
```
